src/utils/credentials-helper.js

```diff
@@ -0,0 +1,244 @@
+  };
+
+  await createDocAclForUser(session, containerUrl, keyAccessAcl);
+  await createDocAclForUser(session, privateKeyUrl, keyAccessAcl);
+  await createDocAclForUser(session, publicKeyUrl, keyAccessAcl);
+};
+
+const getUserSigningKeyInternal = async (session) => {
+  const privateKeyUrl = `${String(
+    session.info.webId.split('profile')[0]
+  )}PASS_Credentials/private_key.ttl`;
+  const dataSet = await getSolidDataset(privateKeyUrl, { fetch: session.fetch });
+  const keyString = getStringNoLocale(
+    getThing(dataSet, `${privateKeyUrl}#privateKey`),
+    RDF_PREDICATES.sha256
+  );
+  const key = await window.crypto.subtle.importKey(
+    'jwk',
+    JSON.parse(keyString),
+    {
+      name: 'RSA-PSS',
+      hash: 'SHA-256'
+    },
+    false,
+    ['sign']
+  );
+  return key;
+};
+
+const InitializePassUserCredentials = async (session) => {
+  const dataSets = await createPassUserCredentials(session);
+  return savePassUserCredentials(dataSets, session);
+};
+
+/**
+ * Fetches user signing key from user's pod. If one does not exist, creates one, then fetches it.
+ *
+ * @memberof utils
+ * @function getUserSigningKey
+ * @param {Session} session - Solid's Session Object (see {@link Session})
+ * @returns {Promise} Promise - File upload is handled via Solid libraries
+ */
+export const getUserSigningKey = async (session) => {
+  let key = null;
+  try {
+    try {
+      key = await getUserSigningKeyInternal(session);
+    } catch (e) {
+      await InitializePassUserCredentials(session);
+      key = await getUserSigningKeyInternal(session);
+    }
+  } catch (e) {
+    key = null;
+  }
+  return key;
+};
+
+/**
+ * Serializes a given Solid DataSet into TTL
+ * Note: Should not be used for networking. Use inrupt's libraries for that
+ *
+ * @memberof utils
+ * @function serializeDataSet
+ * @param {import('@inrupt/solid-client').SolidDataset} dataSet - Inrupt Dataset you want to serialize
+ * @returns {Promise} Promise that resolves to dataset serialized into a ttl string
+ */
+export const serializeDataSet = async (dataSet) => {
+  let serializedTtl = '';
+  const mockFetch = async (_url, { body }) => {
+    serializedTtl = body;
+    return new Response({}, { status: 200 });
+  };
+
+  // Inrupt's Solid client doesn't expose a Thing serializer,
+  // but it does expose a fetch in the saveSolidDatasetInContainer function
+  // that puts the serialized document in the body
+  try {
+    await saveSolidDatasetInContainer('ignored url', dataSet, {
+      fetch: mockFetch
+    });
+  } catch {
+    return serializedTtl;
+  }
+
+  return serializedTtl;
+};
+
+/**
+ * Generates a cryptographic signature for a given document
+ *
+ * @memberof utils
+ * @function generateRsaSignature
+ * @param {CryptoKey} key - The RSA-PSS private key to sign the document with
+ * @param {string} data - The data to generate the signature for
+ * @returns {Promise} Promise that resolves to a string containing a Base64 encoding of the signature
+ */
```

Comment:
Not much of a problem since JSDoc are still just comments at the end of the day, so disregard if you think it's not needed.

For the most part, we have a blank space between the JSDoc comment and the function, would like to keep that consistent.